### DISCUSSION TITLE
How to create arc with CrossSection ?

### DISCUSSION THREAD

**Discussion by rsaccon** [elalish/manifold#605] [Q&A] [answered]
Is there any recommended way to create arcs with CrossSection? With just what CrossSection provides, I can only think of creating them from a polygon.

My use cases: rounding corners of a polygon CrossSection which will be extruded later.

Maybe another way to do it might be to Smooth the Manifold from the the extruded CrossSection, but then I have to turn the Manifold first into a mesh and define the half-edges and I will be far a way from the beautiful high level API the manifold Library provides.

**Comment by elalish** [comment, marked as answer]
`CrossSection` has an `Offset` method where you can choose round corners, so that might help if the arcs have uniform radius. Generally in CSG you make arcs with unions of circles and rectangles. I doubt you'd want `Smooth` as that will leave all your inputs verts unmoved, so it's quite different than rounding off corners.

**Reply by rsaccon** [reply]
That's great and soooo easy ! Thanks a lot.
